keps/sig-storage/3751-volume-provisioned-io/README.md

```diff
@@ -0,0 +1,1338 @@
+
+
+### Test Plan
+
+* Basic unit tests for performance and quota system.
+* API conformance tests
+* E2E tests using mock driver to cause failure on create, update and recovering cases
+  * [K8s storage framework](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/testsuites)
+  * [csi-tes](https://github.com/kubernetes-csi/csi-test)
+* Test coverage of quota usage with ResourceQuota and LimitRange
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature in 1.28 behind the VolumeAttributesClass feature gate which is set to a default of false
+- Add basic unit tests
+
+#### Alpha -> Beta
+- Unit tests and e2e tests outlined in design proposal implemented
+
+#### Beta
+
+- Beta in 1.29: Since this feature is an expansion of the existing capacity usage flow, we are going to move this to beta with enhanced e2e and test coverage 
+
+#### GA
+
+- GA in 1.30, an issue board will be created to track issues, and all major issues should be fixed before GA.
+- No users complaining about the new behavior
+
+### Upgrade / Downgrade Strategy
+
+No change in cluster upgrade / downgrade process.
+
+### Version Skew Strategy
+
+This feature is implemented only in the API server and KCM and controlled by
+`VolumeProvisionedIO` feature gate. Following cases may happen:
+
+| API server | external-resizer | Behavior                                                                                                                         |
+|------------|-----|----------------------------------------------------------------------------------------------------------------------------------|
+| off | off | Existing Kubernetes behavior.                                                                                                    |
+| on | off| Existing Kubernetes behavior, API server gets requests of iops/throughput but does not act on it
+| off | on | external-resizer should not get any event to update iops/throughput, the current resize flow stays the same |
```

Comment:
For this off/on case, what happens there are already `VolumeAttributesClassName` values (and the corresponding attribute objects) written to storage (i.e. from when the feature was previously enabled on the api server)?

Reply:
If there is no change of VolumeAttributesClassName in the PVC, no modify volume call will be triggered. 
But let say if after it is off and then turn it on again, and the end user create a pvc with that pre-exist VolumeAttributesClassName, a modify volume call will be executed.